File: 数据管理/daily_report_jiacheng.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
驾乘险推介率日报生成器

功能：
1. 读取最近14天的数据
2. 按三级机构分组统计：
   - 驾乘险推介率 = 有驾乘险的保单数 / 总保单数 * 100
   - 件均保费 = 总保费 / 总件数
   - 驾乘险件数 = 有驾乘险的保单数
3. 生成结构化日报（结论先行 + 分论点 + 3个表格）
"""

import os
import sys
from datetime import datetime, timedelta
from pathlib import Path
import pandas as pd

# 配置
SCRIPT_DIR = Path(__file__).parent
CONFIG = {
    'data_path': SCRIPT_DIR / 'warehouse/fact/policy/current',
    'output_path': SCRIPT_DIR / '数据分析报告',
    'days_to_analyze': 14,
    'timezone': 'Asia/Shanghai'
}


def find_latest_parquet():
    """查找最新的 Parquet 文件"""
    data_dir = CONFIG['data_path']
    if not data_dir.exists():
        raise FileNotFoundError(f"数据目录不存在: {data_dir}")
    
    parquet_files = list(data_dir.glob('*.parquet'))
    if not parquet_files:
        raise FileNotFoundError(f"未找到 Parquet 文件: {data_dir}")
    
    # 优先选择"每日数据"
    daily_files = [f for f in parquet_files if '每日数据' in f.name]
    if daily_files:
        return sorted(daily_files)[-1]
    
    return sorted(parquet_files)[-1]


def query_data(parquet_path, days=14):
    """查询最近N天的驾乘险数据"""
    # 读取 Parquet 文件
    df = pd.read_parquet(parquet_path)
    
    # 转换日期
    df['签单日期'] = pd.to_datetime(df['签单日期'])
    
    # 筛选最近N天
    end_date = datetime.now().date()
    start_date = end_date - timedelta(days=days)
    
    df = df[
        (df['签单日期'].dt.date >= start_date) &
        (df['签单日期'].dt.date < end_date) &
        (df['是否报价'] == False)  # 排除报价记录
    ]
    
    # 按机构和日期分组统计
    daily_stats = df.groupby([df['三级机构'], df['签单日期'].dt.date]).agg({
        '保单号': 'nunique',
        '交叉销售标识': lambda x: (x == True).sum(),
        '保费': 'sum'
    }).reset_index()
    
    # 重命名列
    daily_stats.columns = ['机构', '日期', '总件数', '驾乘险件数', '总保费']
    
    # 计算推介率和件均保费
    daily_stats['推介率'] = (daily_stats['驾乘险件数'] / daily_stats['总件数'] * 100).round(2)
    daily_stats['件均保费'] = (daily_stats['总保费'] / daily_stats['总件数']).round(2)
    
    # 排序
    daily_stats = daily_stats.sort_values(['机构', '日期'])
    
    return daily_stats.to_dict('records')


def organize_by_org(data):
    """按机构组织数据"""
    org_data = {}
    
    for row in data:
        org = row['机构']
        date = row['日期']
        
        if org not in org_data:
            org_data[org] = {
                'dates': [],
                '推介率': {},
                '件均保费': {},
                '驾乘险件数': {}
            }
        
        if date not in org_data[org]['dates']:
            org_data[org]['dates'].append(date)
        
        org_data[org]['推介率'][date] = row['推介率'] or 0
        org_data[org]['件均保费'][date] = row['件均保费'] or 0
        org_data[org]['驾乘险件数'][date] = int(row['驾乘险件数'] or 0)
    
    # 排序日期
    for org in org_data:
        org_data[org]['dates'].sort()
    
    return org_data


def generate_analysis(org_data):
    """生成分析结论"""
    analyses = []
    orgs = list(org_data.keys())
    
    # 1. 计算整体趋势
    all_dates = set()
    for org in orgs:
        all_dates.update(org_data[org]['dates'])
    sorted_dates = sorted(all_dates)
    
    if len(sorted_dates) >= 7:
        last7_days = sorted_dates[-7:]
        prev7_days = sorted_dates[-14:-7] if len(sorted_dates) >= 14 else []
        
        # 计算最近7天平均值
        last7_avg = {'rate': 0, 'premium': 0, 'count': 0}
        prev7_avg = {'rate': 0, 'premium': 0, 'count': 0}
        
        for org in orgs:
            for date in last7_days:
                if date in org_data[org]['推介率']:
                    last7_avg['rate'] += org_data[org]['推介率'][date]
                    last7_avg['premium'] += org_data[org]['件均保费'][date]
                    last7_avg['count'] += org_data[org]['驾乘险件数'][date]
            
            if prev7_days:
                for date in prev7_days:
                    if date in org_data[org]['推介率']:
                        prev7_avg['rate'] += org_data[org]['推介率'][date]
                        prev7_avg['premium'] += org_data[org]['件均保费'][date]
                        prev7_avg['count'] += org_data[org]['驾乘险件数'][date]
        
        if prev7_days and prev7_avg['rate'] > 0:
            rate_change = last7_avg['rate'] - prev7_avg['rate']
            premium_change = last7_avg['premium'] - prev7_avg['premium']
            count_change = last7_avg['count'] - prev7_avg['count']
            
            analyses.append({
                'type': '整体趋势',
                'content': f"最近7天对比前7天：推介率{'上升' if rate_change > 0 else '下降'} {abs(rate_change):.2f}%，" +
                          f"件均保费{'上升' if premium_change > 0 else '下降'} {abs(premium_change):.0f}元，" +
                          f"驾乘险件数{'增加' if count_change > 0 else '减少'} {abs(count_change)}件"
            })
    
    # 2. 识别问题机构（推介率低于平均水平）
    avg_rates = {}
    for org in orgs:
        rates = list(org_data[org]['推介率'].values())
        avg_rates[org] = sum(rates) / len(rates) if rates else 0
    
    overall_avg = sum(avg_rates.values()) / len(avg_rates) if avg_rates else 0
    problem_orgs = sorted(
        [(org, rate) for org, rate in avg_rates.items() if rate < overall_avg * 0.8],
        key=lambda x: x[1]
    )
    
    if problem_orgs:
        analyses.append({
            'type': '问题机构',
            'content': f"推介率偏低的机构：{', '.join([f'{org}({rate:.2f}%)' for org, rate in problem_orgs])}。" +
                      "建议重点关注培训与激励措施。"
        })
    
    # 3. 识别优秀机构
    top_orgs = sorted(
        [(org, rate) for org, rate in avg_rates.items() if rate > overall_avg * 1.2],
        key=lambda x: x[1],
        reverse=True
    )[:3]
    
    if top_orgs:
        analyses.append({
            'type': '优秀机构',
            'content': f"推介率领先的机构：{', '.join([f'{org}({rate:.2f}%)' for org, rate in top_orgs])}。" +
                      "建议总结推广优秀经验。"
        })
    
    return analyses


def generate_table(title, org_data, metric):
    """生成 Markdown 表格"""
    orgs = sorted(org_data.keys())
    all_dates = set()
    
    for org in orgs:
        all_dates.update(org_data[org]['dates'])
    
    sorted_dates = sorted(all_dates)
    
    # 表头
    table = f"### {title}\n\n"
    table += '| 机构 | ' + ' | '.join([
        f"{d.month}/{d.day}" for d in sorted_dates
    ]) + ' |\n'
    
    table += '|------' + '|------' * len(sorted_dates) + '|\n'
    
    # 表体
    for org in orgs:
        table += f'| {org} |'
        for date in sorted_dates:
            value = org_data[org][metric].get(date)
            if value is not None:
                if metric == '推介率':
                    table += f' {value:.2f}% |'
                elif metric == '件均保费':
                    table += f' {value:.0f} |'
                else:
                    table += f' {int(value)} |'
            else:
                table += ' - |'
        table += '\n'
    
    return table


def generate_report(data, analyses, org_data):
    """生成完整日报"""
    today = datetime.now()
    date_str = today.strftime('%Y年%m月%d日')
    
    report = f"# 驾乘险推介率日报\n\n"
    report += f"**报告日期**: {date_str}\n"
    report += f"**分析周期**: 最近 {CONFIG['days_to_analyze']} 天\n"
    report += f"**数据截止**: {today.strftime('%Y-%m-%d')}\n\n"
    
    report += "---\n\n"
    
    # 一、结论先行
    report += "## 一、核心结论\n\n"
    for idx, analysis in enumerate(analyses, 1):
        report += f"### {idx}. {analysis['type']}\n\n"
        report += f"{analysis['content']}\n\n"
    
    # 二、分论点详解
    report += "## 二、详细分析\n\n"
    
    # 2.1 驾乘险推介率
    report += "### 2.1 驾乘险推介率分析\n\n"
    report += "驾乘险推介率反映了业务员向客户推荐驾乘险的积极性和能力。推介率越高，说明交叉销售工作越到位。\n\n"
    report += "**关键指标**：\n"
    report += "- 推介率 = 驾乘险保单数 / 总保单数 × 100%\n"
    report += "- 目标：推介率应保持在 30% 以上\n"
    report += "- 优秀机构：推介率 > 40%\n\n"
    
    # 2.2 件均保费
    report += "### 2.2 件均保费分析\n\n"
    report += "件均保费反映了车险业务的质量和规模。件均保费高，说明业务员承保的车辆价值较高或保障更全面。\n\n"
    report += "**关键指标**：\n"
    report += "- 件均保费 = 总保费 / 总件数\n"
    report += "- 基准：件均保费 > 3000 元\n"
    report += "- 优质业务：件均保费 > 4000 元\n\n"
    
    # 2.3 驾乘险件数
    report += "### 2.3 驾乘险件数分析\n\n"
    report += "驾乘险件数直接反映了交叉销售的绝对成果。件数越多，说明驾乘险销售业绩越好。\n\n"
    report += "**关键指标**：\n"
    report += "- 驾乘险件数 = 有驾乘险标识的保单数量\n"
    report += "- 日均目标：根据机构规模差异\n"
    report += "- 增长目标：环比增长 > 5%\n\n"
    
    # 三、数据表格
    report += "## 三、详细数据表格\n\n"
    
    report += generate_table('驾乘险推介率 (%)', org_data, '推介率')
    report += '\n\n'
    
    report += generate_table('件均保费 (元)', org_data, '件均保费')
    report += '\n\n'
    
    report += generate_table('驾乘险件数', org_data, '驾乘险件数')
    report += '\n\n'
    
    report += "---\n\n"
    report += f"**报告生成时间**: {today.strftime('%Y-%m-%d %H:%M:%S')}\n"
    report += "**数据来源**: 车险保单综合明细表\n"
    report += "**生成工具**: OpenClaw 自动化日报系统\n"
    
    return report


def main():
    """主函数"""
    print('🚀 驾乘险推介率日报生成器启动...\n')
    
    try:
        # 1. 查找最新数据文件
        print('📂 查找最新数据文件...')
        parquet_path = find_latest_parquet()
        print(f'   ✓ 数据文件: {parquet_path.name}')
        
        # 2. 查询数据
        print(f'📈 查询最近{CONFIG["days_to_analyze"]}天数据...')
        data = query_data(parquet_path, CONFIG['days_to_analyze'])
        print(f'   ✓ 查询到 {len(data)} 条记录')
        
        if not data:
            print('❌ 未查询到数据，请检查数据源')
            sys.exit(1)
        
        # 3. 组织数据
        print('🔄 组织数据结构...')
        org_data = organize_by_org(data)
        print(f'   ✓ {len(org_data)} 个机构')
        
        # 4. 生成分析
        print('💡 生成分析结论...')
        analyses = generate_analysis(org_data)
        
        # 5. 生成报告
        print('📝 生成日报...')
        report = generate_report(data, analyses, org_data)
        
        # 6. 保存报告
        output_dir = CONFIG['output_path']
        output_dir.mkdir(parents=True, exist_ok=True)
        
        output_file = output_dir / f"驾乘险推介率日报_{datetime.now().strftime('%Y-%m-%d')}.md"
        output_file.write_text(report, encoding='utf-8')
        
        print(f'\n✅ 日报生成成功！')
        print(f'📄 文件路径: {output_file}\n')
        
        # 7. 输出到控制台
        print('=' * 80)
        print(report)
        print('=' * 80)
        
        sys.exit(0)
        
    except Exception as e:
        print(f'❌ 生成日报失败: {e}')
        import traceback
        traceback.print_exc()
        sys.exit(1)


if __name__ == '__main__':
    main()

File: 数据管理/daily_report_jiacheng.mjs
#!/usr/bin/env node
/**
 * 驾乘险推介率日报生成器
 * 
 * 功能：
 * 1. 读取最近15天的数据
 * 2. 筛选条件：险种大类 = 商业险
 * 3. 按三级机构分组统计：
 *    - 推介率 = driver_count / auto_count * 100（车架号去重）
 *    - 件均保费 = driver_premium / driver_policy_count
 *    - 驾乘险件数 = driver_policy_count
 * 4. 输出 CSV 格式
 * 
 * 指标口径（与项目一致）：
 * - auto_count: 车险承保车辆数（去重车架号）
 * - driver_count: 驾乘险承保车辆数（去重车架号）
 * - driver_policy_count: 驾乘险保单件数
 * - driver_premium: 驾乘险保费
 * - auto_premium: 车险保费
 */

import duckdb from 'duckdb';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// 配置
const CONFIG = {
  dataPath: path.join(__dirname, 'warehouse/fact/policy/current/每日数据_20250101_20260304.parquet'),
  outputPath: path.join(__dirname, '数据分析报告'),
  daysToAnalyze: 15,  // 改为15天
  timezone: 'Asia/Shanghai'
};

/**
 * 初始化 DuckDB 连接
 */
async function initDB() {
  return new Promise((resolve, reject) => {
    const db = new duckdb.Database(':memory:', (err) => {
      if (err) reject(err);
      else resolve(db);
    });
  });
}

/**
 * 查询最近15天的驾乘险数据
 * 指标口径与项目定义保持一致
 */
async function queryData(db) {
  return new Promise((resolve, reject) => {
    const sql = `
      WITH daily_stats AS (
        SELECT 
          org_level_3 AS "三级机构",
          CAST(policy_date AS DATE) AS "日期",
          
          -- 核心因子（与项目指标定义一致）
          COUNT(DISTINCT vin) AS auto_count,                                    -- 车险承保车辆数（车架号去重）
          COUNT(DISTINCT CASE WHEN is_cross_sell = true THEN vin END) AS driver_count,  -- 驾乘险承保车辆数（车架号去重）
          COUNT(DISTINCT CASE WHEN is_cross_sell = true THEN policy_no END) AS driver_policy_count,  -- 驾乘险保单件数
          SUM(CASE WHEN is_cross_sell = true THEN premium ELSE 0 END) AS driver_premium,  -- 驾乘险保费
          SUM(premium) AS auto_premium,                                          -- 车险保费
          
          -- 计算指标
          ROUND(
            COUNT(DISTINCT CASE WHEN is_cross_sell = true THEN vin END) * 100.0 / 
            NULLIF(COUNT(DISTINCT vin), 0), 
            2
          ) AS "推介率",
          
          ROUND(
            SUM(CASE WHEN is_cross_sell = true THEN premium ELSE 0 END) / 
            NULLIF(COUNT(DISTINCT CASE WHEN is_cross_sell = true THEN policy_no END), 0),
            2
          ) AS "驾乘险件均保费",
          
          ROUND(
            SUM(premium) / NULLIF(COUNT(DISTINCT vin), 0),
            2
          ) AS "车险件均保费"
          
        FROM read_parquet('${CONFIG.dataPath}')
        WHERE policy_date >= CURRENT_DATE - INTERVAL '${CONFIG.daysToAnalyze} days'
          AND policy_date < CURRENT_DATE
          AND is_quote = false          -- 排除报价记录
          AND coverage_type = '商业险'  -- 只筛选商业险
        GROUP BY org_level_3, CAST(policy_date AS DATE)
      )
      
      SELECT 
        "三级机构",
        "日期",
        auto_count,
        driver_count,
        driver_policy_count,
        driver_premium,
        auto_premium,
        "推介率",
        "驾乘险件均保费",
        "车险件均保费"
      FROM daily_stats
      ORDER BY "三级机构", "日期"
    `;
    
    db.all(sql, (err, rows) => {
      if (err) reject(err);
      else resolve(rows);
    });
  });
}

/**
 * 按机构组织数据（用于表格生成）
 */
function organizeByOrg(data) {
  const orgData = {};
  
  data.forEach(row => {
    const org = row['三级机构'];
    const date = row['日期'];
    
    if (!orgData[org]) {
      orgData[org] = {
        dates: [],
        推介率: {},
        驾乘险件均保费: {},
        driver_policy_count: {}
      };
    }
    
    if (!orgData[org].dates.includes(date)) {
      orgData[org].dates.push(date);
    }
    
    orgData[org].推介率[date] = row['推介率'] || 0;
    orgData[org].驾乘险件均保费[date] = row['驾乘险件均保费'] || 0;
    orgData[org].driver_policy_count[date] = row['driver_policy_count'] || 0;
  });
  
  // 排序日期
  Object.keys(orgData).forEach(org => {
    orgData[org].dates.sort();
  });
  
  return orgData;
}

/**
 * 生成分析结论
 */
function generateAnalysis(orgData) {
  const analyses = [];
  const orgs = Object.keys(orgData);
  
  // 1. 计算整体趋势
  let overallTrends = {
    avgRecommendRate: [],
    avgPremium: [],
    totalCount: []
  };
  
  // 获取最近7天和前7天的对比数据
  const allDates = new Set();
  orgs.forEach(org => {
    orgData[org].dates.forEach(d => allDates.add(d));
  });
  const sortedDates = Array.from(allDates).sort();
  
  if (sortedDates.length >= 7) {
    const last7Days = sortedDates.slice(-7);
    const prev7Days = sortedDates.slice(-14, -7);
    
    // 计算最近7天平均值
    let last7Avg = { rate: 0, premium: 0, count: 0 };
    let prev7Avg = { rate: 0, premium: 0, count: 0 };
    
    orgs.forEach(org => {
      last7Days.forEach(date => {
        if (orgData[org].推介率[date] !== undefined) {
          last7Avg.rate += orgData[org].推介率[date];
          last7Avg.premium += orgData[org].驾乘险件均保费[date];
          last7Avg.count += orgData[org].driver_policy_count[date];
        }
      });
      
      prev7Days.forEach(date => {
        if (orgData[org].推介率[date] !== undefined) {
          prev7Avg.rate += orgData[org].推介率[date];
          prev7Avg.premium += orgData[org].驾乘险件均保费[date];
          prev7Avg.count += orgData[org].driver_policy_count[date];
        }
      });
    });
    
    // 生成趋势分析
    const rateChange = (last7Avg.rate - prev7Avg.rate).toFixed(2);
    const premiumChange = (last7Avg.premium - prev7Avg.premium).toFixed(2);
    const countChange = last7Avg.count - prev7Avg.count;
    
    analyses.push({
      type: '整体趋势',
      content: `最近7天对比前7天：推介率${rateChange > 0 ? '上升' : '下降'} ${Math.abs(rateChange)}%，` +
               `驾乘险件均保费${premiumChange > 0 ? '上升' : '下降'} ${Math.abs(premiumChange)}元，` +
               `驾乘险件数${countChange > 0 ? '增加' : '减少'} ${Math.abs(countChange)}件`
    });
  }
  
  // 2. 识别问题机构（推介率低于平均水平）
  let avgRates = {};
  orgs.forEach(org => {
    const rates = Object.values(orgData[org].推介率);
    avgRates[org] = rates.reduce((a, b) => a + b, 0) / rates.length;
  });
  
  const overallAvg = Object.values(avgRates).reduce((a, b) => a + b, 0) / orgs.length;
  const problemOrgs = Object.entries(avgRates)
    .filter(([org, rate]) => rate < overallAvg * 0.8) // 低于平均80%
    .sort((a, b) => a[1] - b[1]);
  
  if (problemOrgs.length > 0) {
    analyses.push({
      type: '问题机构',
      content: `推介率偏低的机构：${problemOrgs.map(([org, rate]) => 
        `${org}(${rate.toFixed(2)}%)`
      ).join('、')}。建议重点关注培训与激励措施。`
    });
  }
  
  // 3. 识别优秀机构
  const topOrgs = Object.entries(avgRates)
    .filter(([org, rate]) => rate > overallAvg * 1.2)
    .sort((a, b) => b[1] - a[1])
    .slice(0, 3);
  
  if (topOrgs.length > 0) {
    analyses.push({
      type: '优秀机构',
      content: `推介率领先的机构：${topOrgs.map(([org, rate]) => 
        `${org}(${rate.toFixed(2)}%)`
      ).join('、')}。建议总结推广优秀经验。`
    });
  }
  
  return analyses;
}

/**
 * 生成 Markdown 表格
 */
function generateTable(title, orgData, metric, isPercent = false) {
  const orgs = Object.keys(orgData).sort();
  const allDates = new Set();
  
  orgs.forEach(org => {
    orgData[org].dates.forEach(d => allDates.add(d));
  });
  
  const sortedDates = Array.from(allDates).sort();
  
  // 表头
  let table = `### ${title}\n\n`;
  table += '| 三级机构 | ' + sortedDates.map(d => {
    const date = new Date(d);
    return `${date.getMonth() + 1}/${date.getDate()}`;
  }).join(' | ') + ' |\n';
  
  table += '|------' + '|------'.repeat(sortedDates.length) + '|\n';
  
  // 表体
  orgs.forEach(org => {
    table += `| ${org} |`;
    sortedDates.forEach(date => {
      const value = orgData[org][metric][date];
      if (value !== undefined) {
        if (isPercent) {
          table += ` ${value.toFixed(2)}% |`;
        } else if (metric === '驾乘险件均保费') {
          table += ` ${value.toFixed(0)} |`;
        } else {
          table += ` ${value} |`;
        }
      } else {
        table += ' - |';
      }
    });
    table += '\n';
  });
  
  return table;
}

/**
 * 生成 CSV 数据（用于数据分析）
 */
function generateCSV(data) {
  // CSV 表头
  let csv = '三级机构,日期,auto_count,driver_count,driver_policy_count,driver_premium,auto_premium,推介率,驾乘险件均保费,车险件均保费\n';
  
  // CSV 数据行
  data.forEach(row => {
    csv += `"${row['三级机构']}",${row['日期']},${row.auto_count},${row.driver_count},${row.driver_policy_count},${row.driver_premium},${row.auto_premium},${row['推介率']},${row['驾乘险件均保费']},${row['车险件均保费']}\n`;
  });
  
  return csv;
}

/**
 * 生成完整日报（Markdown）
 */
function generateReport(data, analyses, orgData) {
  const today = new Date();
  const dateStr = today.toLocaleDateString('zh-CN', { 
    year: 'numeric', 
    month: '2-digit', 
    day: '2-digit',
    timeZone: CONFIG.timezone
  });
  
  let report = `# 驾乘险推介率日报\n\n`;
  report += `**报告日期**: ${dateStr}\n`;
  report += `**分析周期**: 最近 ${CONFIG.daysToAnalyze} 天\n`;
  report += `**数据截止**: ${new Date().toISOString().split('T')[0]}\n`;
  report += `**筛选条件**: 险种大类 = 商业险\n\n`;
  
  report += `---\n\n`;
  
  // 一、结论先行
  report += `## 一、核心结论\n\n`;
  analyses.forEach((analysis, idx) => {
    report += `### ${idx + 1}. ${analysis.type}\n\n`;
    report += `${analysis.content}\n\n`;
  });
  
  // 二、分论点详解
  report += `## 二、详细分析\n\n`;
  
  // 2.1 驾乘险推介率
  report += `### 2.1 驾乘险推介率分析\n\n`;
  report += `驾乘险推介率反映了业务员向客户推荐驾乘险的积极性和能力。推介率越高，说明交叉销售工作越到位。\n\n`;
  report += `**关键指标**：\n`;
  report += `- 推介率 = driver_count / auto_count × 100%\n`;
  report += `- 目标：推介率应保持在 30% 以上\n`;
  report += `- 优秀机构：推介率 > 40%\n\n`;
  
  // 2.2 驾乘险件均保费
  report += `### 2.2 驾乘险件均保费分析\n\n`;
  report += `驾乘险件均保费反映了驾乘险保单的价值。件均保费高，说明业务员推荐的驾乘险保障更全面。\n\n`;
  report += `**关键指标**：\n`;
  report += `- 驾乘险件均保费 = driver_premium / driver_policy_count\n`;
  report += `- 基准：件均保费 > 200 元\n\n`;
  
  // 2.3 驾乘险件数
  report += `### 2.3 驾乘险件数分析\n\n`;
  report += `驾乘险件数直接反映了交叉销售的绝对成果。件数越多，说明驾乘险销售业绩越好。\n\n`;
  report += `**关键指标**：\n`;
  report += `- 驾乘险件数 = driver_policy_count\n`;
  report += `- 增长目标：环比增长 > 5%\n\n`;
  
  // 三、数据表格
  report += `## 三、详细数据表格\n\n`;
  
  report += generateTable('驾乘险推介率 (%)', orgData, '推介率', true);
  report += '\n\n';
  
  report += generateTable('驾乘险件均保费 (元)', orgData, '驾乘险件均保费');
  report += '\n\n';
  
  report += generateTable('驾乘险件数', orgData, 'driver_policy_count');
  report += '\n\n';
  
  report += `---\n\n`;
  report += `**报告生成时间**: ${new Date().toLocaleString('zh-CN', { timeZone: CONFIG.timezone })}\n`;
  report += `**数据来源**: 车险保单综合明细表（商业险）\n`;
  report += `**生成工具**: OpenClaw 自动化日报系统\n`;
  
  return report;
}

/**
 * 主函数
 */
async function main() {
  console.log('🚀 驾乘险推介率日报生成器启动...\n');
  
  try {
    // 1. 初始化数据库
    console.log('📊 初始化 DuckDB...');
    const db = await initDB();
    
    // 2. 查询数据
    console.log(`📈 查询最近${CONFIG.daysToAnalyze}天数据（商业险）...`);
    const data = await queryData(db);
    console.log(`   ✓ 查询到 ${data.length} 条记录`);
    
    if (data.length === 0) {
      console.error('❌ 未查询到数据，请检查数据源');
      process.exit(1);
    }
    
    // 3. 组织数据
    console.log('🔄 组织数据结构...');
    const orgData = organizeByOrg(data);
    console.log(`   ✓ ${Object.keys(orgData).length} 个三级机构`);
    
    // 4. 生成分析
    console.log('💡 生成分析结论...');
    const analyses = generateAnalysis(orgData);
    
    // 5. 生成报告
    console.log('📝 生成日报...');
    const report = generateReport(data, analyses, orgData);
    
    // 6. 生成 CSV 数据
    console.log('📊 生成 CSV 数据...');
    const csv = generateCSV(data);
    
    // 7. 保存文件
    const dateStr = new Date().toISOString().split('T')[0];
    const reportPath = path.join(CONFIG.outputPath, `驾乘险推介率日报_${dateStr}.md`);
    const csvPath = path.join(CONFIG.outputPath, `驾乘险推介率数据_${dateStr}.csv`);
    
    fs.mkdirSync(CONFIG.outputPath, { recursive: true });
    fs.writeFileSync(reportPath, report, 'utf-8');
    fs.writeFileSync(csvPath, csv, 'utf-8');
    
    console.log(`\n✅ 日报生成成功！`);
    console.log(`📄 Markdown: ${reportPath}`);
    console.log(`📊 CSV数据: ${csvPath}\n`);
    
    // 8. 输出到控制台
    console.log('='.repeat(80));
    console.log(report);
    console.log('='.repeat(80));
    
    // 9. 关闭数据库
    db.close();
    
    process.exit(0);
  } catch (error) {
    console.error('❌ 生成日报失败:', error);
    process.exit(1);
  }
}

// 执行主函数
main();
